sdk/react/panels/CastDevicesScreen.js

```diff
@@ -0,0 +1,159 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Animated,
+  ListView,
+  Text,
+  View,
+  TouchableHighlight
+} from 'react-native';
+
+import {
+  BUTTON_NAMES
+} from '../constants';
+import Utils from '../utils';
+
+import castScreenStyles from './style/castScreenStyles.json';
+const styles = Utils.getStyles(castScreenStyles);
+
+let ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 === r2});
+
+const dismissButtonSize = 20;
+const castButtonSize = 35;
+
+class CastDevicesScreen extends Component {
+  static propTypes = {
+    height: PropTypes.number,
+    width: PropTypes.number,
+    onDismiss: PropTypes.func,
+    onDeviceSelected: PropTypes.func,
+    config: PropTypes.object,
+    deviceIds: PropTypes.array,
+    deviceNames: PropTypes.array,
+    selectedItem: PropTypes.string
+  };
+
+  componentWillMount() {
+    this.state = {
+      translateY: new Animated.Value(this.props.height),
```

Comment:
why keep functions in the state instead of class methods?

Reply:
see comments above